### PR TITLE
Pin urllib3 for E2E tests

### PR DESCRIPTION
**Pull request by achocron**
vcrpy is incompatible with urllib3 2.0.2, so this pins urllib3 to an older version.

### CODE REVIEW

File: azure-quantum/tests.live/Run.ps1 [hunk 1]
```diff
@@ -105,6 +105,10 @@ if ($PackageDir -Match "azure-quantum") {
     Copy-Item -Path (Join-Path $PackageDir "tests" "unit" "*.qs") -Destination $PSScriptRoot
 }
 
+# Pin urllib3 for compatibility with vcrpy
+Write-Host "##[info]Install version of urllib3 that is compatible with vcrpy"
+pip install urllib3==1.26.15
```
Comment (ashwinmayya) [inline azure-quantum/tests.live/Run.ps1:110]: Could this be in the requirements.txt file instead?

Comment (ashwinmayya) [reply azure-quantum/tests.live/Run.ps1:110]: Or some other requirements file that is used during build

**Review by shenyingjun — APPROVED** [review]

File: azure-quantum/tests.live/Run.ps1 (hunk 1 above)
Comment (achocron) [reply azure-quantum/tests.live/Run.ps1:110]: The requirements files seem to apply to not just test environments. This seemed to by the only test-specific place to include the pin.

**Comment by vxfield** [comment]
Closed as superseded by #468

**Pull request closed without merging**